esphome/components/json/json_util.cpp

```diff
@@ -4,6 +4,9 @@
 #ifdef USE_ESP8266
 #include <Esp.h>
 #endif
+#ifdef USE_ESP_IDF
```

Comment:
I think `USE_ESP32` is more appropriate here, since that's also what's checked before using this constant.
```suggestion
#ifdef USE_ESP32
```

Reply:
actually only  idf needs the include. Arduino (esp32) has the include already somewhere in their code
I followed the pattern from https://github.com/esphome/esphome/blob/a02d2e2e113df3bef6fc69c8929784d9b039b509/esphome/components/debug/debug_component.cpp#L7

Reply:
> actually only idf needs the include. Arduino (esp32) has the include already somewhere in their code

Yes, but I prefer being explicit about importing all the headers we need, so that we don't break if Arduino moves a header include around (this'll actually get enforced by CI in the future).

> I followed the pattern from

Note that in that file the usage is guarded by `USE_ESP_IDF` as well, while here the usage is guarded by `USE_ESP32`.

Reply:
good point and doesn't harm anyways